Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM node:17.4.0-buster as node-builder
+RUN cp -r /usr/local/bin /usr/local/cp-bin \
+  && rm /usr/local/cp-bin/corepack /usr/local/cp-bin/docker-entrypoint.sh
+
+FROM ruby:3.0.3-buster as base
+
+# install nodejs
+RUN mkdir /usr/local/lib/node_modules /opt/yarn-v1.22.17
+COPY --from=node-builder /usr/local/lib/node_modules /usr/local/lib/node_modules
+COPY --from=node-builder /opt/yarn-v1.22.17 /opt/yarn-v1.22.17
+COPY --from=node-builder /usr/local/cp-bin /usr/local/bin
+
+ENV DATABASE_PASSWORD=''
+
+RUN set -ex; \
+    curl https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - ; \
+    echo "deb http://apt.postgresql.org/pub/repos/apt/ buster-pgdg main 14" > /etc/apt/sources.list.d/pgdg.list; \
+    apt-get update -y; \
+    apt-get install -y --no-install-recommends \
+    postgresql-14 postgresql-client-14; \
+    echo 'host    all             all             127.0.0.1/32            trust' > /etc/postgresql/14/main/pg_hba.conf
+
+WORKDIR /myapp
+
+COPY Gemfile Gemfile.lock /myapp/
+
+RUN gem install bundler \
+  && bundle config set --local without test \
+  && bundle install -j4
```

Comment:
You're right. I'll remove unnecessary commands.

Reply:
Did you remove the `bundle config` command for some reason? I was playing around with this and I noticed, that with this configuration, the app was failing to boot for me. I think because of a spring issue, fixed by https://github.com/rails/spring/commit/7fcb7427204b0873bc32799930e7b58ff2c7e71a.

Reply:
I updated both bundler & spring separately and if you rebase I think this should cause no issues, so we can restore the `bundle config set --local without test`, which by the way can be just `bundle config set without test` because `--local` is now the default. All this assuming that's why you removed it.

Reply:
I removed config set because error occurs when `rails db:setup` command is executed.
After rebase error has been resolved, so I'll revert `bundle config set` setting.

Reply:
Yeah, that's the issue I was having and it should be fixed by #597!